script.js

```diff
@@ -1,4 +1,89 @@
+//Player Name
+var playerName = " ";
+var mode = "input username";
+
+// Player Win Loss Draw number
+var playerWins = 0;
+var playerLosses = 0;
+var playerDraws = 0;
+
+// Players Inputs
+var scissors = "scissors";
+var paper = "paper";
+var stone = "stone";
+
+//computer randomly choose scissors paper or stone
+var getRandomNum = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomNum = randomInteger + 1;
+  return randomNum;
+};
+var assignedNumToWord = function () {
+  var randomNum = getRandomNum();
+  if (randomNum == 1) return scissors;
+  if (randomNum == 2) return paper;
+  if (randomNum == 3) return stone;
+};
+
+// Rules: scissors beats paper, paper beats stone, and stone beats scissors. If same object draw
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (mode == "input username") {
+    playerName = input;
+    mode = "play sps";
+    return (
+      "Welcome " + playerName + ", please enter scissors, paper or stone. "
+    );
+  }
+  if (mode == "play sps") {
+    var computersRandomWord = assignedNumToWord();
```

Comment:
Good use of a helper function to abstract the logic out of `main`. This helps make `main` easier to read.